addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -729,93 +729,50 @@ registry['sizing_y'] = registry.sizing.extend({
  */
 registry.colorpicker = SnippetOption.extend({
     xmlDependencies: ['/web_editor/static/src/xml/snippets.xml'],
-    events: _.extend({}, SnippetOption.prototype.events || {}, {
-        'click .colorpicker button': '_onColorButtonClick',
-        'mouseenter .colorpicker button': '_onColorButtonEnter',
-        'mouseleave .colorpicker button': '_onColorButtonLeave',
-        'click .note-color-reset': '_onColorResetButtonClick',
-    }),
-    colorPrefix: 'bg-',
+    custom_events: {
+        color_picked: '_onColorPicked',
+        custom_color_picked: '_onCustomColor',
+        color_hover: '_onColorHovered',
+        color_leave: '_onColorLeft',
+        color_reset: '_onColorReset',
+    },
 
     /**
      * @override
      */
     start: function () {
-        var self = this;
-        var res = this._super.apply(this, arguments);
-
-        if (this.data.colorPrefix) {
-            this.colorPrefix = this.data.colorPrefix;
+        let prom = [];
+        prom.push(this._super.apply(this, arguments));
+        let excluded;
+        if (this.data.paletteExclude) {
+            excluded = this.data.paletteExclude.replace(/ /g, '').split(',');
         }
+        this.colorPalette = new ColorPaletteWidget(this, {
+            $target: this.$target,
+            colorPrefix: this.data.colorPrefix || undefined,
+            excluded: excluded,
+        });
+        prom.push(this.colorPalette.appendTo(this.$el.find('we-collapse')));
```

Comment:
It will not be a collapse but a select. And you can use the `buildSelectElement` method to build it.

Reply:
There is a second commit